cvat/apps/engine/task.py

```diff
@@ -48,22 +50,43 @@ def rq_handler(job, exc_type, exc_value, traceback):
 ############################# Internal implementation for server API
 
 class _FrameExtractor:
-    def __init__(self, source_path, compress_quality, flip_flag=False):
+    def __init__(self, source_path, compress_quality, step=1, start=0, stop=0, flip_flag=False):
         # translate inversed range 1:95 to 2:32
         translated_quality = 96 - compress_quality
         translated_quality = round((((translated_quality - 1) * (31 - 2)) / (95 - 1)) + 2)
+        self.source = source_path
         self.output = tempfile.mkdtemp(prefix='cvat-', suffix='.data')
         target_path = os.path.join(self.output, '%d.jpg')
         output_opts = '-start_number 0 -b:v 10000k -vsync 0 -an -y -q:v ' + str(translated_quality)
+        filters = ''
+        if stop > 0:
+            filters = 'between(n,' + str(start) + ',' + str(stop) + ')'
+        elif start > 0:
+            filters = 'gte(n,' + str(start) + ')'
+        if step > 1:
+            filters += ('*' if filters else '') + 'not(mod(n,' + str(step) + '))'
+        if filters:
+            filters = "select=\"'" + filters + "'\""
         if flip_flag:
-            output_opts += ' -vf "transpose=2,transpose=2"'
+            filters += (',' if filters else '') + 'transpose=2,transpose=2'
+        if filters:
+            output_opts += ' -vf ' + filters
         ff = FFmpeg(
             inputs  = {source_path: None},
             outputs = {target_path: output_opts})
 
         slogger.glob.info("FFMpeg cmd: {} ".format(ff.cmd))
         ff.run()
 
+    def getframenumber(self):
```

Comment:
@zliang7 , are you sure that the code works in all cases (I don't see `-vsync 0` option)? Can we instead get all files inside output directory with frame*.jpg pattern, sort them and extract number of frames from the latest file?